File: src/main/kotlin/ru/havlong/cards/api/dto/SampleDTO.java
package ru.havlong.cards.api.dto;

import io.swagger.annotations.ApiModel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.havlong.cards.store.entity.Sample;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ApiModel(description = "Describes sample.")
public class SampleDTO {

    private Long id;

    public static SampleDTO of(Sample ent) {

        return SampleDTO.builder()
                .id(ent.getId())
                .build();
    }

    public static List<SampleDTO> of(Stream<Sample> sampleStream) {

        return sampleStream
                .map(SampleDTO::of)
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/ru/havlong/cards/transfer/SampleDTO.kt
package ru.havlong.cards.transfer

import io.swagger.annotations.ApiModel
import ru.havlong.cards.models.Sample

/**
 * 20.06.2020
 * SampleDTO
 *
 * @author Havlong
 * @version v1.0
 */
@ApiModel(description = "Model description here")
data class SampleDTO(val id: Long?) {
    companion object {
        @JvmStatic
        fun of(entity: Sample): SampleDTO = SampleDTO(entity.id)

        @JvmStatic
        fun of(samples: List<Sample>): List<SampleDTO> = samples.map(Companion::of)
    }
}
